app/shared/src/forms/TextInput.js

```diff
@@ -2,17 +2,84 @@
 
 import * as React from 'react';
 import { TextInput as OriginalTextInput, View } from 'react-native';
+import { Translation } from '@kiwicom/mobile-localization';
 
 import StyleSheet from '../PlatformStyleSheet';
 import Icon from '../icons/Icon';
 import Color from '../Color';
-import type { StylePropType } from '../../types/Styles';
+import Text from '../Text';
 
-const styles = StyleSheet.create({
+type Props = {|
+  placeholder?: React.Element<typeof Translation>,
+  value?: string,
+  autoFocus?: boolean,
+  iconName?: string, // only Material icons allowed here
+  onChangeText?: (text: string) => void,
+  keyboardType?: 'email-address',
+  secureTextEntry?: boolean,
+|};
+
+type State = {|
+  displayPlaceholder: boolean,
+|};
+
+export default class TextInout extends React.Component<Props, State> {
```

Comment:
I rewrote this component so we can use `Translation` in the `placeholder` prop.

Reply:
Do you think you could also rename it to `TextInput`?

Reply:
Yes, this is not intentional... :)